libraries/chain/webassembly/runtimes/eos-vm-oc/executor.cpp

```diff
@@ -102,25 +103,42 @@ static void throw_internal_exception(const char* const s) {
 	void name()
 
 DEFINE_EOSVMOC_TRAP_INTRINSIC(eosvmoc_internal,depth_assert) {
-   throw_internal_exception("Exceeded call depth maximum");
+   throw_internal_exception(wasm_execution_error(FC_LOG_MESSAGE(error, "Exceeded call depth maximum")));
 }
 
 DEFINE_EOSVMOC_TRAP_INTRINSIC(eosvmoc_internal,div0_or_overflow) {
-   throw_internal_exception("Division by 0 or integer overflow trapped");
+   throw_internal_exception(wasm_execution_error(FC_LOG_MESSAGE(error, "Division by 0 or integer overflow trapped")));
 }
 
 DEFINE_EOSVMOC_TRAP_INTRINSIC(eosvmoc_internal,indirect_call_mismatch) {
-   throw_internal_exception("Indirect call function type mismatch");
+   throw_internal_exception(wasm_execution_error(FC_LOG_MESSAGE(error, "Indirect call function type mismatch")));
 }
 
 DEFINE_EOSVMOC_TRAP_INTRINSIC(eosvmoc_internal,indirect_call_oob) {
-   throw_internal_exception("Indirect call index out of bounds");
+   throw_internal_exception(wasm_execution_error(FC_LOG_MESSAGE(error, "Indirect call index out of bounds")));
 }
 
 DEFINE_EOSVMOC_TRAP_INTRINSIC(eosvmoc_internal,unreachable) {
-   throw_internal_exception("Unreachable reached");
+   throw_internal_exception(wasm_execution_error(FC_LOG_MESSAGE(error, "Unreachable reached")));
 }
 
+static void eos_vm_oc_check_memcpy_params(int32_t dest, int32_t src, int32_t length) {
+   //make sure dest & src are zexted when converted from signed 32-bit to signed ptrdiff_t; length should always be small but do it too
+   const unsigned udest = dest;
+   const unsigned usrc = src;
+   const unsigned ulength = length;
+
+   //this must remain the same behavior as the memcpy host function
+   if((size_t)(std::abs((ptrdiff_t)udest - (ptrdiff_t)usrc)) >= ulength)
```

Comment:
This check compiles down to about ~11 instructions but can still consume a substantial amount of CPU (beyond 1% in some cases). Perhaps something more clever can shave off another 0.5%.

Reply:
> This check compiles down to about ~11 instructions but can still consume a substantial amount of CPU (beyond 1% in some cases). Perhaps something more clever can shave off another 0.5%.

Did you try plain `if` comparisons without using `std::abs`?